arch/xtensa/core/xtensa-asm2-util.S

```diff
@@ -265,6 +265,30 @@ DEF_EXCINT 6, _handle_excint, xtensa_int6_c
 DEF_EXCINT 7, _handle_excint, xtensa_int7_c
 #endif
 
+/* The user exception vector is defined here, as we need to handle
+ * MOVSP exceptions in assembly (the result has to be to unspill the
+ * caller function of the code that took the exception, and that can't
+ * be done in C).  A prototype exists which mucks with the stack frame
+ * from the C handler instead, but that would add a LARGE overhead to
+ * some alloca() calls (those whent he caller has been spilled) just
+ * to save these five cycles during other exceptions and L1
+ * interrupts.  Maybe revisit at some point, with better benchmarking.
```

Comment:
In commit message you mention some possible future optimizations. I suggest filing a low-priority user story about it so we don't forget. We could give it to a junior engineer or something.